src/auth/dto/authcreds.dto.ts

```diff
@@ -0,0 +1,15 @@
+import { IsString, IsNotEmpty, IsEmail } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class AuthCredsDto {
+  @IsString()
+  @ApiProperty({ description: 'The email of the user' })
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @IsString()
+  @ApiProperty({ description: 'The password of the user' })
+  @IsNotEmpty()
+  password: string;
```

Comment:
PullProofAI

The `AuthCredsDto` class looks well-defined with proper validation decorators and Swagger annotations for email and password fields. It follows best practices by using `IsString`, `IsEmail`, and `IsNotEmpty` decorators for validation.

This code snippet is well-structured, readable, and maintains clear separation of concerns by handling validation within the DTO class itself. Great job on adhering to validation standards and enhancing API documentation through Swagger annotations.


<!-- This is an auto-generated comment by PullProofAI -->